common/djangoapps/student/views.py

```diff
@@ -2377,3 +2380,34 @@ def _get_course_programs(user, user_enrolled_courses):  # pylint: disable=invali
                 log.warning('Program structure is invalid, skipping display: %r', program)
 
     return programs_data
+
+
+def _get_programs_credentials(user):
```

Comment:
This is really `_get_xseries_credentials`, isn't it? I'd change the names of variables here and in the dashboard template to reflect that.